torch_xla/csrc/tensor_methods.cpp

```diff
@@ -621,13 +621,24 @@ XLATensor XLATensor::cast(const XLATensor& input, at::ScalarType dtype) {
 XLATensor XLATensor::cat(tensorflow::gtl::ArraySlice<const XLATensor> tensors,
                          xla::int64 dim) {
   XLA_CHECK_GT(tensors.size(), 0);
+  xla::Shape template_shape;
```

Comment:
What error are we trying to address here?
Can you post which case within that test fails?

Reply:
`cat` has a special logic when to throw errors... The test failure is still valid. I'll send a commit later.